engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -4,99 +4,90 @@
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.security.Policy;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.Enumeration;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static com.google.common.base.Verify.verifyNotNull;
-
 public class ModuleManager {
-    /** Set this environment variable to "true" to load all modules in the classpath by default. */
-    public final static String LOAD_CLASSPATH_MODULES_ENV = "TERASOLOGY_LOAD_CLASSPATH_MODULES";
+    /** Set this property to "true" to allow modules on the classpath. */
     public final static String LOAD_CLASSPATH_MODULES_PROPERTY = "org.terasology.load_classpath_modules";
 
     private static final Logger logger = LoggerFactory.getLogger(ModuleManager.class);
     private final StandardPermissionProviderFactory permissionProviderFactory = new StandardPermissionProviderFactory();
-    private final PermissionProviderFactory wrappingPermissionProviderFactory = new WarnOnlyProviderFactory(permissionProviderFactory);
+    private final PermissionProviderFactory wrappingPermissionProviderFactory =
+            new WarnOnlyProviderFactory(permissionProviderFactory);
 
     private final ModuleRegistry registry;
     private ModuleEnvironment environment;
     private final ModuleMetadataJsonAdapter metadataReader;
+    private final ModuleFactory moduleFactory;
     private final ModuleInstallManager installManager;
-    private Module engineModule;
+    private final Module engineModule;
 
     public ModuleManager(String masterServerAddress) {
         this(masterServerAddress, Collections.emptyList());
     }
 
     public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspathsToAddToEngine) {
-        this(masterServerAddress, classesOnClasspathsToAddToEngine, null);
-    }
-
-    public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspathsToAddToEngine, Boolean loadModulesFromClasspath) {
         PathManager pathManager = PathManager.getInstance();  // get early so if it needs to initialize, it does it now
 
         metadataReader = newMetadataReader();
 
-        engineModule = loadEngineModule(classesOnClasspathsToAddToEngine);
+        if (Boolean.getBoolean(LOAD_CLASSPATH_MODULES_PROPERTY)) {
+            moduleFactory = new ClasspathCompromisingModuleFactory();
+        } else {
+            moduleFactory = new ModuleFactory();
+        }
+        moduleFactory.setDefaultLibsSubpath("build/libs");
+
+        Map<String, ModuleMetadataLoader> mmlm = moduleFactory.getModuleMetadataLoaderMap();
+        mmlm.put(TerasologyConstants.MODULE_INFO_FILENAME.toString(), metadataReader);
+        mmlm.put("engine-module.txt", metadataReader);  // FIXME: this should be *only* for engine-module.
+        // FIXME: …or maybe engine-module doesn't need a special filename anymore?
```

Comment:
FIXME: engine-module.txt should only be for engine module, if at all.

Reply:
Is `engine-module.txt` treated any different anywhere? We're obviously using the same reader here, at least...

Reply:
I _imagine_ that engine has this `engine-module.txt` has this unique filename because it was loaded from a `class.getResource` call https://github.com/MovingBlocks/Terasology/blob/0fa1272241b43be94e9920c84a41683643c9f240/engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java#L128  
instead of a filesystem API that opens a file in the root of a specific directory or archive.

So maybe it needed a name to distinguish it from a `/module.txt` file that might have been put there by anything else on the classpath?

but now that we've moved it to `org/terasology/engine/` maybe that's enough of a qualifier that it could just be named `module.txt` again?